src/trio/_core/_tests/test_parking_lot.py

```diff
@@ -249,30 +261,81 @@ async def test_parking_lot_breaker_basic() -> None:
     # breaking the lot again with the same task is a no-op
     lot.break_lot()
 
-    # but with a different task it gives a warning
-    async def dummy_task(
-        task_status: _core.TaskStatus[_core.Task] = trio.TASK_STATUS_IGNORED,
-    ) -> None:
-        task_status.started(_core.current_task())
+    child_task = None
+    with pytest.warns(RuntimeWarning):
+        async with trio.open_nursery() as nursery:
+            child_task = await nursery.start(dummy_task)
+            # registering a task as breaker on an already broken lot is fine... though it
+            # maybe shouldn't be as it will always cause a RuntimeWarning???
+            # Or is this a sign that we shouldn't raise a warning?
+            add_parking_lot_breaker(child_task, lot)
+            nursery.cancel_scope.cancel()
+
+    # manually breaking a lot with an already exited task is fine
+    lot = ParkingLot()
+    lot.break_lot(child_task)
+
+
+async def test_parking_lot_breaker_warnings() -> None:
+    lot = ParkingLot()
+    task = current_task()
+    lot.break_lot()
 
+    warn_str = "attempted to break parking .* already broken by .*"
+    # breaking an already broken lot with a different task gives a warning
     # The nursery is only to create a task we can pass to lot.break_lot
-    # and has no effect on the test otherwise.
     async with trio.open_nursery() as nursery:
         child_task = await nursery.start(dummy_task)
         with pytest.warns(
             RuntimeWarning,
-            match="attempted to break parking .* already broken by .*",
+            match=warn_str,
         ):
             lot.break_lot(child_task)
         nursery.cancel_scope.cancel()
 
+    # note that this get put into an exceptiongroup if inside a nursery, making any
+    # stacklevel arguments irrelevant
+    with RaisesGroup(Matcher(RuntimeWarning, match=warn_str)):
+        async with trio.open_nursery() as nursery:
+            child_task = await nursery.start(dummy_task)
+            lot.break_lot(child_task)
+            nursery.cancel_scope.cancel()
+
     # and doesn't change broken_by
     assert lot.broken_by == task
 
+    # register multiple tasks as lot breakers, then have them all exit
+    lot = ParkingLot()
+    child_task = None
+    # This does not give an exception group, as the warning is raised by the nursery
+    # exiting, and not any of the tasks inside the nursery.
+    # And we only get a single warning because... of the default warning filter? and the
+    # location being the same? (because I haven't figured out why stacklevel makes no
+    # difference)
```

Comment:
Yep, [once per location](https://docs.python.org/3/library/warnings.html#warning-filter) is the default, and `with pytest.warns()` clears preexisting filters.